src/main/java/com/google/sps/data/User.java

```diff
@@ -1,57 +1,195 @@
+  private static String CLIENT_ID = "1034390229233-u07o0iaas2oql8l4jhe7fevpfsbrtv7n.apps.googleusercontent.com";
+  // Payload class contains decrypted user information.
+  private Payload payload;
+  private Entity entity;
 
-  public User(String id, String email, String username) {
-    this.id = id;
-    this.email = email;
-    this.username = username;
-    planner = new ArrayList<Long>();
-    cookbook = new ArrayList<Long>();
-    userRecipes = new ArrayList<Long>();
+  /**
+    // User Entity Properties:
+
+    "displayName" : "Example Name"
+    "cookbook" : [ "longId1", "longId2", "longId3" ]
+    "userRecipes" : [ "longId1", "longId2", "longId3" ]
+    "planner" : [ "longId1", "longId2", "longId3" ]
+    "shoppingList" : [ "longId1", "longId2", "longId3" ]
+
+    // Properties that can be accessed from Payload object:
+
+    // These seven fields are only included when the user has granted the "profile" and
+    // "email" OAuth scopes to the application.
+    // "sub" or Subject refers to the user-unique ID number.
+    "sub" : "110169484474386276334"
+    "email": "testuser@gmail.com",
+    "email_verified": "true",
+    "name" : "Test User",
+    "picture": "https://lh4.googleusercontent.com/-kYgzyAWpZzJ/ABCDEFGHI/AAAJKLMNOP/tIXL9Ir44LE/s99-c/photo.jpg",
+    "given_name": "Test",
+    "family_name": "User",
+    "locale": "en"
+  */
+
+  /**
+   * Prevents usage of default constructor
+   */
+  private User() { }
+
+  /**
+   * Constructs a User instance based on a unique id_token.
+   * Verify user using google's ID token verifyer API.
+   * This method is commented out because it does not work yet.
+   */
+  public User(String idTokenString) {
+    // GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier
+    // .Builder(UrlFetchTransport.getDefaultInstance(), new JsonFactory())
+    // // Specify the CLIENT_ID of the app that accesses the backend:
+    // .setAudience(Collections.singletonList(CLIENT_ID)).build();
+
+    // GoogleIdToken idToken = verifier.verify(idTokenString);
+    // if (idToken == null) {
+    //   throw new Exception("ID token verification failed");
+    // }
+
+    // payload = idToken.getPayload();
+  }
+
+  /**
+   * This creates a standard universal user instance regardless of login status.
+   * This test method will not be removed from the webapp prior to deployment.
+   */
+  public static User getTestUser() {
+    User testUser = new User();
+    String testID = "000000000000000000000";
+    Key userKey = KeyFactory.createKey("User", testID);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    try {
+        testUser.entity = datastore.get(userKey);
+    } catch(Exception e) {
+        // On EntityNotFoundException, create initial instance.
+        testUser.entity = new Entity(userKey);
+        String emptyJsonArray = "[ ]";
+        testUser.entity.setProperty("displayName", "Test Person");
+        testUser.entity.setProperty("cookbook", emptyJsonArray);
```

Comment:
I'm not sure what the scope of your testing will be, but if you set all these properties to point to the same string, it may be the case that they will all end up pointing to the same reference, and a change in one of the properties would be reflected in all of them. Additionally, if your test user is storing string data in entity properties which are expected to have ArrayList<String> data everywhere else, you may run in to unexpected behavior/errors as you try to use this code later on to test different parts of the website.

Reply:
The entity is a datastore Entity,  Entity.setProperty copies the string literal "[ ]" and does not hold any reference.
Also, the String Json arrays are fully in abstraction, the get and add methods control conversion to and from ArrayList data, and no one will handle the data without them.
I have tested the functionality of the methods using a local test Servlet.

Reply:
SGTM then!